OpenSim/Actuators/Millard2012EquilibriumMuscle.h

```diff
@@ -712,11 +703,13 @@ OpenSim_DECLARE_CONCRETE_OBJECT(Millard2012EquilibriumMuscle, Muscle);
 // PRIVATE UTILITY CLASS MEMBERS
 //==============================================================================
 
-    // This object defines the pennation model used for this muscle model. Using
-    // a ClonePtr saves us from having to write a destructor, copy constructor,
-    // or copy assignment. This will be zeroed on construction, cleaned up on
-    // destruction, and cloned when copying.
-    MuscleFixedWidthPennationModel penMdl;
+    // Subcomponents owned by the muscle. The properties of these subcomponents
+    // are set in extendFinalizeFromProperties() from the properties of the
+    // muscle.
+    MemberSubcomponentIndex penMdlIdx{
+      constructSubcomponent<MuscleFixedWidthPennationModel>("penMdl") };
```

Comment:
I have not claimed that the design is "desirable", but there were several practical reasons for making these private subcomponents and not removing the original properties of Millard2012EquilibriumMuscle; here are two:

1. The design philosophy of the Millard2012EquilibriumMuscle class is focused around a particular type of muscle model (one fiber, one tendon, etc.). The muscle can operate in a few configurations and its curves can be adjusted, but the class was not designed in a modular fashion. I think that modifying this class would be more work than creating a new model that has been designed to be modular.
2. Tests use `Object::RenameType()`, which assumes that the properties of Millard2012EquilibriumMuscle have the same names as those of Thelen2003Muscle. I disagree with using `Object::RenameType()` in tests, and there should be exceptions thrown if unexpected xml elements are encountered during deserialization; however, changing the tests and deserializer would be more work than is reasonable for this PR.

This PR improves how properties are handled; it is not a redesign of the muscle.